### DISCUSSION TITLE
Solokey v2 is not seen/recognised by Firefox Mobile

### DISCUSSION THREAD

**Discussion by axelsimon** [solokeys/solo2#150] [Q&A] [answered]
Using Firefox Mobile (strictly speaking, Firefox Nightly), i Just tried to log in to my Github account using a USB-C Solokey linked to my account as a security key. I plugged it in to my phone's USB-C port and touched the connectors: nothing happens.

I'm not sure if this is an issue on the Solo side, or on the Firefox Mobile side. I'll be reporting this on that end too.

**Comment by iGadget** [comment]
Have you tested other browsers? Are they working as expected?

**Reply by alexjmoore** [reply]
The keys, for me, simply don't work on android, NFC or usb, any browser.

**Reply by axelsimon** [reply]
> Have you tested other browsers? Are they working as expected?

Just tried Vanadium, GrapheneOS's version of Chromium, and i immediately get "Security key authentication failed" when trying to use they, both with it plugged in and not plugged in.

**Reply by rpavlik** [reply]
yeah I can't get mine to work on Android at all either.

**Reply by iGadget** [reply]
I just tested logging in to my Nextcloud instance with Firefox mobile 109.2.0 on a Fairphone 4 (Android 11). Solo2 USB-C (original firmware) works as expected.
So apparently it's not the browser(s). Which Android versions are you running?

**Reply by alexjmoore** [reply]
In my case I'm running Android 13 on a Pixel 7 Pro, but it also failed on a Pixel 4 XL running Android 12.

**Reply by iGadget** [reply]
I don't have access to any devices with Android 12 (yet), unfortunately. What could also be an issue is (the lack of?) Google Play services. At least on my desktop, I noticed Ungoogled Chromium would not work with the Solo2 whatsoever. Perhaps a similar mechanism could be at work with ROMs shipping a stripped down / altered version of Google Play services or even completely without them?

**Reply by Hetti** [reply]
I just tested and realized that it seems that pixel phones do not deliver enough power via NFC to be recognized properly by the phone, if the key is powered externally via USB it gets recognized via NFC.

**Reply by axelsimon** [reply]
This seemed like a great idea, so i tried it but no luck, it makes no difference. In fact, the key is recognised over NFC, but scanning it just offers to open www.solokeys.com.

On the [Mozilla bugtracker](https://bugzilla.mozilla.org/show_bug.cgi?id=1808057), i've been told that Firefox Mobile uses GMS (Google Mobile Settings) to implement WebAuthN. As (my version of) Graphene comes without any Google services, not even MicroG, that probably explains why the whole thing fails.

My current guess/understanding is that since here is no support at the OS level to know what to do with the WebAuthN over NFC offered by the Solokey, so it just degrades to offering a plaintext URL.

**Comment by axelsimon** [comment, marked as answer]
The Solokey is not recognised for WebAuthN because Firefox [depends](https://bugzilla.mozilla.org/show_bug.cgi?id=1808057#c3) on [Google Mobile Services](https://developers.google.com/android/reference/com/google/android/gms/auth/package-summary), which isn't included in my version of Android.
